### PR TITLE
[rfr] Contributor order

### PR DESCRIPTION
**Pull request by sloria**
Implements contributor ordering.

Tested by https://github.com/sloria/osf.io/commit/dc17ca0a1f7e4cb22aff08f4e4be70b5d1ea6727

### CODE REVIEW

**Comment by sloria** [comment]
@cwisecarver By default, contributors will be ordered by insertion time of the corresponding `Contributor` records. I am unsure if this will affect your migrations or not.

**Comment by cwisecarver** [comment]
Thanks for the heads up @sloria. I haven't gotten to contributors yet, they'll have to be special for other reasons so this shouldn't be a big deal. I'm going to head out a little early. I'll be back online in 45m or so and I'll review this and try to finish up the migration stuff for the models. It's so close but I just keep hitting weird data issues with the newish models.

File: osf_models/models/node.py [hunk 1]
```diff
@@ -54,6 +54,10 @@ class AbstractNode(TypedModel, AddonModelMixin, IdentifierMixin,
     #: Whether this is a pointer or not
     primary = True
 
+    FIELD_ALIASES = {
```
Comment (cwisecarver) [inline osf_models/models/node.py]: Good pattern. Maybe we should TODO the other places where we're using the _* pattern to revisit.

File: osf_models/models/node.py [hunk 2]
```diff
@@ -81,9 +85,15 @@ class AbstractNode(TypedModel, AddonModelMixin, IdentifierMixin,
...
     # TODO: Can this be a reference instead of data?
     child_node_subscriptions = DateTimeAwareJSONField(default=dict, blank=True)
-    contributors = models.ManyToManyField(OSFUser,
+    _contributors = models.ManyToManyField(OSFUser,
                                           through=Contributor,
                                           related_name='nodes')
+
+    @property
+    def contributors(self):
+        # NOTE: _order field is generated by order_with_respect_to = 'node'
```
Comment (cwisecarver) [inline osf_models/models/node.py]: Might mention this is set on the Contributor model in the note for people unfamiliar with Django.

File: osf_models/models/node.py [hunk 3]
```diff
@@ -946,6 +956,9 @@ def manage_contributors(self, user_dicts, auth, save=False):
                 )
 
             if to_retain != users:
+                # TODO: Can we prevent n queries?
+                sorted_contribs = [self.contributor_set.get(user=user).pk for user in users]
```
Comment (cwisecarver) [inline osf_models/models/node.py]: Would the order be kept if you did this:
```python
user_pks = [user.pk for user in users]
sorted_contribs = self.contributor_set.filter(user_pk__in=user_pks).values_list('pk', flat=True)
```

**Comment by cwisecarver** [comment]
Looks like some tests are still referencing `contributors` and they're failing but this PR lgtm. I didn't functionally test it but I'm sure it'll be fine once the tests are passing.

File: osf_models/models/node.py (hunk 3 above)
Comment (sloria) [inline osf_models/models/node.py]: Tried that and it didn't work. =/

Comment (cwisecarver) [inline osf_models/models/node.py]: Ah, that sucks.

**Pull request merged**